server/app/services/program/ProgramService.java

```diff
@@ -877,13 +919,22 @@ public ErrorAnd<ProgramDefinition, CiviFormError> editStatus(
             : ImmutableList.copyOf(statusesCopy);
 
     program.statusDefinitions().setStatuses(updatedStatuses);
-
-    return ErrorAnd.of(
-        syncProgramDefinitionQuestions(
-                programRepository.getShallowProgramDefinition(
-                    programRepository.updateProgramSync(program.toProgram())))
-            .toCompletableFuture()
-            .join());
+    ErrorAnd<ProgramDefinition, CiviFormError> result =
+        ErrorAnd.of(
+            syncProgramDefinitionQuestions(
+                    programRepository.getShallowProgramDefinition(
+                        programRepository.updateProgramSync(program.toProgram())))
+                .toCompletableFuture()
+                .join());
+    if (!result.isError()) {
+      // Temporary hook to create a new row in ApplicationStatuses table for the status change.
```

Comment:
**question:** Same question as above.

Reply:
Will be moved soon. So leaving it as such :)